datumaro/plugins/tiling/merge_tile.py

```diff
@@ -0,0 +1,303 @@
+    Tile and MergeTile. For example, Tile -> (an arbitrary Transform) -> MergeTile.
+    """
+
+    _merge_anns_func_map: Dict[AnnotationType, Callable[..., List[Annotation]]] = {
+        AnnotationType.label: _merge_by_copy,
+        AnnotationType.mask: _merge_mask,
+        AnnotationType.points: _merge_points,
+        AnnotationType.polygon: _merge_polygon,
+        AnnotationType.polyline: _merge_polyline,
+        AnnotationType.bbox: _merge_bbox,
+        AnnotationType.caption: _merge_by_copy,
+        AnnotationType.cuboid_3d: _merge_not_support,
+        AnnotationType.super_resolution_annotation: _merge_not_support,
+        AnnotationType.depth_annotation: _merge_depth_annotation,
+    }
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        return parser
+
+    def __init__(self, extractor):
+        super().__init__(extractor)
+
+    def __iter__(self):
+        items_to_merge = defaultdict(list)
+
+        for item in self._extractor:
+            item_id = item.attributes.get("tile_id")
+            roi = item.attributes.get("roi")
+
+            if item_id is not None and roi is not None:
+                items_to_merge[item_id] += [item]
+
+        for item_id, items in items_to_merge.items():
+            yield self._merge_items(item_id, items)
+
+    def _merge_items(self, item_id: str, items: List[DatasetItem]) -> DatasetItem:
+        assert len(items) > 0
+
+        max_h = 0
+        max_w = 0
+        for item in items:
+            roi = item.attributes.get("roi")
+            x, y, w, h = roi
+            max_w = max(max_w, x + w)
+            max_h = max(max_h, y + h)
+        img_size = (max_h, max_w)
+
+        merged_item = self.wrap_item(
+            items[0],
+            id=item_id,
+            media=MosaicImage(
+                [
+                    (
+                        item.media,
+                        item.attributes.get("roi"),
+                    )
+                    for item in items
+                ],
+                img_size,
+            ),
+            attributes=self._merge_tiled_attributes(items),
+            annotations=self._merge_tiled_annotations(items, img_size),
+        )
+
+        return merged_item
+
+    @staticmethod
+    def _merge_tiled_attributes(items: List[DatasetItem]) -> Dict[str, Any]:
+        attrs = {}
+        for item in items:
+            attrs.update(item.attributes)
+
+        del attrs["tile_idx"]
+        del attrs["tile_id"]
+        del attrs["roi"]
+        return attrs
+
+    def _merge_tiled_annotations(
+        self, items: List[DatasetItem], img_size: Tuple[int, int]
+    ) -> List[Annotation]:
+        anns_to_merge: Dict[AnnotationType, AnnotationsForMerge] = defaultdict(list)
+
+        for item in items:
+            roi = item.attributes.get("roi")
+            roi_box: sg.Polygon = sg.box(*xywh_to_x1y1x2y2(*roi))
```

Comment:
Under the consideration of not-perfectly divided tiles, e.g., divide a length 10 with a width 4, isn't it better to use x1y1x2y2 instead of xywh? in this case, rois (x,w) will be (0,4), (4,8), and (8,2)?

Reply:
Sorry. I can't figure out what your claim is. Aren't `x1y1x2y2` and `xywh` formats perfectly convertible each other? In this line, I just convert `xywh(roi)` to `x1y1x2y2` to create `sg.Polygon` which only allows `x1y1x2y2` format.